demo/demo_sat.py

```diff
@@ -85,24 +110,28 @@ def __init__(self, name="Space Oddity"):
                 Simulates achieving an RF Lock with the spacecraft.
                 """
                 await asyncio.sleep(2)
+                self.check_cancelled(id=command.id, major_tom=major_tom)
```

Comment:
I guess this is the best way with asyncio code. If this were a Thread, cancellation could raise an exception inside of it. There is probably a way to take the Task returned by `ensure_future` and cancel it as well.

Reply:
This is a thread, since it never awaits anything. Once the command callback is called, it steps through single-threaded (except for the sleeps). The check cancel function raises an specific exception that I catch and use to cancel. 

But yeah this is a pretty naive implementation, but it works! Also the reality is most commands can't be cancelled (really only uplink_file or something like that with lots of ground processing). I think it's more appropriate for the command thread to check at the points it can actually be cancelled.